File: CS/XtraTabWithButtons/Paint/SkinTabPainterDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraTab.Drawing;
using DevExpress.XtraTab;
using DevExpress.XtraTab.ViewInfo;
using System.Drawing;

namespace XtraTabWithButtons
{
    public class SkinTabPainterDescendant : SkinTabPainter
    {
        public SkinTabPainterDescendant(IXtraTab tabControl)
            : base(tabControl) { }

        protected override void DrawHeaderPage(TabDrawArgs e, BaseTabRowViewInfo row, BaseTabPageViewInfo pInfo)
        {
            base.DrawHeaderPage(e, row, pInfo);
            BaseTabPageViewInfoDescendant pInfodes = pInfo as BaseTabPageViewInfoDescendant;
            DrawHeaderPageButtons(e, pInfodes);
        }

        protected virtual void DrawHeaderPageButtons(TabDrawArgs e, BaseTabPageViewInfoDescendant pInfo)
        {
            pInfo.PagePanel.Draw(e.Cache);
        }
    }
}

File: VB/XtraTabWithButtons/Paint/SkinTabPainterDescendant.vb
Imports DevExpress.XtraTab.Drawing
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.ViewInfo
Imports System.Drawing

Namespace XtraTabWithButtons

    Public Class SkinTabPainterDescendant
        Inherits SkinTabPainter

        Public Sub New(ByVal tabControl As IXtraTab)
            MyBase.New(tabControl)
        End Sub

        Protected Overrides Sub DrawHeaderPage(ByVal e As TabDrawArgs, ByVal row As BaseTabRowViewInfo, ByVal pInfo As BaseTabPageViewInfo)
            MyBase.DrawHeaderPage(e, row, pInfo)
            Dim pInfodes As BaseTabPageViewInfoDescendant = TryCast(pInfo, BaseTabPageViewInfoDescendant)
            DrawHeaderPageButtons(e, pInfodes)
        End Sub

        Protected Overridable Sub DrawHeaderPageButtons(ByVal e As TabDrawArgs, ByVal pInfo As BaseTabPageViewInfoDescendant)
            pInfo.PagePanel.Draw(e.Cache)
        End Sub
    End Class
End Namespace
